src/k8s/cmd/k8s/k8s_bootstrap.go

```diff
@@ -47,6 +48,24 @@ func newBootstrapCmd(env cmdutil.ExecutionEnvironment) *cobra.Command {
 		Long:   "Generate certificates, configure service arguments and start the Kubernetes services.",
 		PreRun: chainPreRunHooks(hookRequireRoot(env), hookInitializeFormatter(env, &opts.outputFormat), hookCheckLXD()),
 		Run: func(cmd *cobra.Command, args []string) {
+			snapdClient, err := snapd.NewClient()
+			if err != nil {
+				cmd.PrintErrln("Error: failed to create snapd client: %w", err)
+				env.Exit(1)
+				return
+			}
+			microk8sInfo, err := snapdClient.GetSnapInfo("microk8s")
+			if err != nil {
+				cmd.PrintErrln("Error: failed to check if microk8s is installed: %w", err)
+				env.Exit(1)
+				return
+			}
+			if microk8sInfo.StatusCode == 200 {
+				cmd.PrintErrln("Error: microk8s snap is installed, please remove it and try again.")
```

Comment:
Just as a suggestion, maybe we can also guide the user on how to do that? e.g. "... By running `snap remove microk8s`"

Reply:
Sure